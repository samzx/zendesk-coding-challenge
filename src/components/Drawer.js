import React from 'react';

import Pagination from './Pagination';
import List from './List';

import { url, tickets } from '../app';

const initialPage = 1;

class Drawer extends React.Component {

    state = {
        currentPage: 0,
        loading: false,
    }

    fetchListings = (page) => {
        if(this.state.loading || this.state.currentPage === page) return;
        this.setState({ currentPage: page, loading: true });
        fetch(url + tickets + `/${page}`, {
            method: 'GET',
        })
        .then((res) => {
            if(res.ok) {
                return res.json();
            } else {
                throw res;
            }
        })
        .then((json) => {
            this.setState({ loading: false });
            this.props.setData(json);
        })
        .catch((e) => {
            this.setState({ errors: `${e.status} ${e.statusText}`, loading: false });
            console.log(e);
        });
    }

    componentDidMount() {
        this.fetchListings(initialPage);
    }

    render() {
        const { errors, loading } = this.state;
        return(
            <div className="drawer">
            { 
                loading &&
                <div className="loading" >
                    <p>{"Loading..."}</p> 
                </div>
            }
            {
                errors && <p className="loading" >{errors}</p>
            }
                <div className="banner" />
                <List
                    data={this.props.data}
                    currentPage={this.state.currentPage}
                    currentTicket={this.props.currentTicket}
                    setTicket={this.props.setTicket}
                    loading={this.state.loading}
                />
                <Pagination
                    data={this.props.data}
                    currentPage={this.state.currentPage}
                    fetchListings={this.fetchListings}
                />
            </div>
        );
    }
}

export default Drawer;