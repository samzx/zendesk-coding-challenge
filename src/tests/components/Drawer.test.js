import React from "react";
import { shallow } from "enzyme";
import Drawer from "../../components/Drawer";

// Using jest-fetch-mock
describe("<Drawer />", () => {
  it("should render Drawer correctly", () => {
    // const wrapper = shallow(<Drawer />);
    // expect(wrapper).toMatchSnapshot();
  });
})
