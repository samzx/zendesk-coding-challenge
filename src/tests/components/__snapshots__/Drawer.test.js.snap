// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Drawer correctly 1`] = `
<div
  className="drawer"
>
  <div
    className="message"
  >
    <p>
      Loading...
    </p>
  </div>
  <div
    className="banner"
  />
  <List
    currentPage={1}
    loading={true}
  />
  <Pagination
    currentPage={1}
    fetchListings={[Function]}
  />
</div>
`;
